### ISSUE TITLE
[Question] Colab usage limitation

### ISSUE THREAD

**Issue by bbalanazi** [Surrey-EEEM071-CVDL/CourseWork#1]
dear i have trained the model until it reaches to 54 and the goole colab is shutdown I have to do the train again for Resent50 ,
can you please show me how i can save the trained model and reopen again?

**Comment by cf00573** [comment]
Hi, I am having the same issue. The code runs for about 6 hours and then disconnects before the training is complete for the second time. It now can't connect to any GPU and is asking me to pay for more due to usage limits. Is there a university version we can use or will I have to pay for colab pro on my personal account?

**Comment by enisb00** [comment]
Yep same here, I don't feel like it is appropriate that we have to essentially abuse a free service to train a network for 6-8 hours (and 5-10 times with different parameters/models) to be able to complete our coursework. The timeout seems to be anywhere from a day to potentially 5 days until you're allowed to use a GPU again which is also not really feasible when it comes to coursework.

It wouldn't be as great of an issue if the example code given in this repo implemented checkpoints (unless it does and I'm not aware of it) allowing us to restart the training without losing all the calculated weights in the previous Colab session.

**Comment by BrandonHanx** [comment]
Hi guys, thanks for your questions.

First of all, please bear in mind that we don't encourage you to achieve a SOTA performance on this task -- so the **absolute performance doesn't matter.**
We just want you to engage in a pipeline of deep learning projects.
You just need to try something you think might be helpful to the performance and report your observations to us (eg, how and why it should work) -- so the **relative performance matters**.

All the default settings (archs, hyper-parameters) in this ref repo are just for your reference, they are not the best proper ones for this coursework. You can choose a lightweight baseline and verify your ideas on top of it. To design such a baseline with a much faster training speed, you can:
1. Use a smaller CNN, like ResNet18
2. Decrease the number of training epoch
3. Split out a smaller training set

If the Colab interrupts your experiments, you can use --resume=[save checkpoint] in the command line to resume the training process.

Also, there are some hack solutions for keeping Colab alive, like https://stackoverflow.com/questions/57113226/how-can-i-prevent-google-colab-from-disconnecting . And a naive solution is to register another Google account...

Finally, please understand the GPU limitation is a typical problem for every deep learning researcher (even for us PhD guys...). Please try the solutions I mentioned above first to see if you can manage to run several exps. If you still suffer from it, we will consider using our university's servers. But those servers are also very crowded and even worse than Colab since you need to set up conda envs and so on. We'll take your difficulties into account when grading your reports, so don't worry.

**Comment by BrandonHanx** [comment]
guys, I updated the codebase with the default settings changed, please refer to https://github.com/Surrey-EEEM071-CVDL/CourseWork/commit/72b2f37debf0c58333d4dc824244d1a380a8bdd2

Specifically, changed arch to mobilenetv3_small, changed epochs to 30 but increased batch_size to 256,
now one training with the default recipe spends 20mins on my single 2080Ti (about 30-40mins on Colab)

**Comment by ehb00** [comment]
Thanks for the update, in the last couple days I've also swapped to ResNet18 and it's definitely more reasonable to train as by my estimations it would take ~1:30hrs on my 2080 for 60 epochs (probably excessive but was the previous default).

Just as an extra note: I do understand it's really hard to arrange compute resources for 100-150 students for a single month to do this coursework but doing this coursework becomes significantly easier if you have hardware at home to the point where I feel that students are at a big disadvantage if they are forced to use Colab as you can't just leave it to work in the background even if it does take 30-40 mins or overnight (hence why I've given up on it and would rather pay the extra electricity costs at home but not everyone can do this).

**Comment by BrandonHanx** [comment]
@enisb00 Thanks for your understanding. Another reason for choosing Colab is because it can be used out-of-box. This year we have about 150 students with very mixed backgrounds. Using univ's GPUs can raise many problems for students who didn't have Linux experience -- so we encourage students to use Colab to focus more on the project itself rather than engineering things. Anyway, we are always trying to adapt this coursework according to your feedback since this is a 100% new coursework, so please just let me know if you have more problems.

**Comment by alex-mcdermott** [comment]
Hi, the VeRi dataset can't be downloaded at the moment. Is there a way you could get another link for it or something please?
<img width="903" alt="Screenshot 2023-04-22 at 5 45 44 pm" src="https://user-images.githubusercontent.com/109023345/233796630-de14e0c0-2c97-4a62-8d86-4fe4ac442f60.png">

**Comment by BrandonHanx** [comment]
Please open the link the msg suggests and save the dataset to your own Google Drive.

Then you can copy this dataset from /myDrive to /content in your colab.
